pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -258,6 +262,10 @@ func (h *upgradeHandler) OnChanged(_ string, upgrade *harvesterv1.Upgrade) (*har
 			return h.upgradeClient.Update(toUpdate)
 		}
 
+		if err := h.reenableAddons(upgrade); err != nil {
```

Comment:
I noticed we have two times to re-enable the descheduler.
Is this not enough if we only re-enable with the corresponding operation during the upgrade successed?

If so, maybe we could remove the re-enable on the cleanup caller.